### PR TITLE
[EG-65] Improved error reporting on stdout

### PR DESCRIPTION
**Pull request by woggioni**
I modified the `ErrorCodeRewritePolicy` to concatenate all the error messages of the exception's cause chain.
The code will start from the throwable being passed to the logger and concatenate its error message with the one in its cause and proceed recursively until it finds an exception with no cause or it detects a loop (an exception already encountered in the traversal).

This should provide customers with more information about errors without havingg to look at the logs (that should still remain the primary source of information)

Here is an example of a stdout after a faiulure opening a JDBC connection with Postgres before the change
```
[ERROR] 10:41:00+0000 [main] pool.HikariPool. - HikariPool-1 - Exception during pool initialization. [errorCode=7gemva, moreInformationAt=https://errors.corda.net/OS/4.3-SNAPSHOT/7gemva]
[ERROR] 10:41:00+0000 [main] internal.NodeStartupLogging. - Exception during node startup: net.corda.nodeapi.internal.persistence.CouldNotCreateDataSourceException: Could not connect to the database. Please check your JDBC connection URL, or the connectivity to the database. [errorCode=1a3cce8, moreInformationAt=https://errors.corda.net/OS/4.3-SNAPSHOT/1a3cce8]
```
And here is the same situation after this change
```
[ERROR] 15:27:14+0000 [main] pool.HikariPool. - Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections. - Connection refused (Connection refused) [errorCode=7gemva, moreInformationAt=https://errors.corda.net/OS/4.3-SNAPSHOT/7gemva]
[ERROR] 15:27:14+0000 [main] internal.NodeStartupLogging. - Could not connect to the database. Please check your JDBC connection URL, or the connectivity to the database. - Failed to initialize pool: Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections. - Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections. - Connection refused (Connection refused) [errorCode=gn3vsm, moreInformationAt=https://errors.corda.net/OS/4.3-SNAPSHOT/gn3vsm]
```

### CODE REVIEW

**Review by JamesHR3 — COMMENTED** [review]
I think this looks reasonable. Is it possible to add a small unit test for this? (I'm aware that testing in common/logging is lacking at the moment) Mostly just to prove that we can't get into a loop.

File: common/logging/src/test/kotlin/net/corda/commmon/logging/WalkExceptionCausedByListTest.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                            e = TestThrowable(i, stack.lastOrNull())
+                            stack.add(e!!)
+                        }
+                        field.set(stack[0], stack[4])
+                        arrayOf("Exception with loop in cause list", e!!, stack.asReversed())
+                    })
+        }
+    }
+
+    @Test
```
Comment (JamesHR3) [inline common/logging/src/test/kotlin/net/corda/commmon/logging/WalkExceptionCausedByListTest.kt]: Might want to stick the `timeout=300_000` in this annotation as we're going to start enforcing it at some point soon.

**Review by JamesHR3 — CHANGES_REQUESTED** [review]
Really minor comment then I think we're good to go.

**Review by JamesHR3 — APPROVED** [review]
LGTM!

**Review by fenryka — APPROVED** [review]
LGTM

**Pull request merged**